src/FSharp.Core/array.fs

```diff
@@ -2,6 +2,8 @@
 
 namespace Microsoft.FSharp.Collections
 
+#nowarn "1118" // 'Make' marked 'inline', perhaps because a recursive value was marked 'inline'
```

Comment:
What exactly is it warning about here?

Reply:
@vzarytovskii 
Okay to explain:
array.fs contains this code:
````
        let private createPartitionsUpToWithMinChunkSize maxIdxExclusive minChunkSize (array: 'T array) =
            [|
            //  A bunch of uninteresting code elided
            |]

        let private createPartitionsUpTo maxIdxExclusive (array: 'T array) =
            createPartitionsUpToWithMinChunkSize maxIdxExclusive minChunkSize array

        let inline groupByImplParallel
            (comparer: IEqualityComparer<'SafeKey>)
            ([<InlineIfLambda>] keyf: 'T -> 'SafeKey)
            ([<InlineIfLambda>] getKey: 'SafeKey -> 'Key)
            (array: 'T array)
            =
            //  A bunch of uninteresting code elided
            let inputChunks = createPartitionsUpTo array.Length array

            //  A bunch of uninteresting code elided
````

It should be noted that: createPartitionsUpToWithMinChunkSize  and createPartitionsUpToWithMinChunkSize, this means they are not accessible outside of the class in which they are defined.  Which is okay in F# code, because we are all in the same class.  However: groupByImplParallel is exposed to a public Api in a different module via inlining.  The F# compiler always errors when an out of scope Api (private) is included in an inline function.

The realsig+ feature will treat these 'private' apis as private, realsig+ will maintain the current shipped behaviour and treat them as internal.

And so ... **inline groupByImplParallel** is out of scope in realsig+, whereas it is in-scope in realsig-.  So the change I have made is to change the error to a warning, so that developers can take a look and modify or optimize their code differently, and not inline createPartitionsUpTo.  It seems that this is the right approach, because, not inlining retains the same semantic meaning as the previous version, and the possible performance degrade is notified.

This happens a few places in Array.parallel so rather than just making the privates internal, I chose to turn off the warning, however, @T-Gro convinced me to change the privates to internal, so the optimization can remain.

I hope this clears things up.

Kevin